_posts/2024-09-27-learning-a-programming-language-with-generative-ai.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: Learning a programming language with generative ai
+author: andandersson
+excerpt: >
+  Experiences from learning a programing language with generatve ai.
+tags:
+ - Generative ai
+ - C#
+ - .net
+---
+
+# Background
+
+Generative artificial intelligence, or GenAI, is one of the biggest hypes in the software industry. With the introduction of applications like ChatGPT, AI-algorithms can produce text, generate images and even motion pictures and music. Due to the trends, there have been many discussions how GenAI may affect software development. How can it be utilized,for example for coding, learning new tools and frameworks, code reviewing? 
+
+![A genAI-created picture of a robot coding](/img/2024-09-27-learning-a-programming-language-with-generative-ai/coding-robot.png)
+
+In the GenAI-twin project, we wanted to find out these things. For the period of March-September 2024, we started to develop a full stack application in C# with .NET and React with the help of GenAI. In that way, we were able to get a picture of how GenAI can affect software development.
+
+As a part of the journey, we also wanted to find out how GenAI can be utilized for learning a new programming language or framework. Myself as a junior developer, I did not have any prior knowledge of neither C# nor .NET and was assigned with the task.
+
+Even though there are several GenAI-tools for coding, I decided to use ChatGPT since the purpose was to learn, rather than just develop. GenAI-tools for coding, like Cursor or Codeium, are used for giving suggestions for code rather than explain concepts and theory.
+With ChatGPT, I could ask questions about the code as well as the theory behind C# and .NET.
+
+# The first application
+
+My first task was to make a very simple full stack-application for the purpose of learning the basics of C# and .NET. The method I used was to tell ChatGPT small tasks at a time, that I wanted to implement. For example, my very first comment was “I am a software developer that don't know anything about neither C# or .net, and I want to learn both by creating a simple application”. The process started, and ChatGPT suggested that I could create an online shop, which I did. I used the same chat window during the whole process, and step by step I asked questions about implementing different features as well as error handling. When needed, I asked about concepts and theory behind C#. 
+
+In general, the results were okay. A working application was created almost completely with the use of GenAI. At some points I had to google certain things, but ChatGPT made most of the work. The quality of the code was varying, although it mainly worked. Some methods were not the most effective possible, naming conventions were not followed, and other minor issues. 
```

Comment:
"The quality of the code was varying, although it mainly worked" -> "The quality of the code varied, although mostly it worked."

Reply:
corrected